tsconfig.dev.json

```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "module": "ESNext",
     "moduleResolution": "Bundler",
-    "noEmit": true
+    "noEmit": true,
+    "noUncheckedIndexedAccess": false,
```

Comment:
why is this false here? 🤔

Reply:
`tsconfig.dev` is used for tests where we don't really need to enforce this rule

Reply:
wouldn't it be still good to have tests with no unchecked index access? can see it infact being more useful because it would spot errors before execution level :D

Reply:
not sure, let's discuss

I can enable it for tests but in that case many utilities we write would need to be compliant with more eager checks 
I am in favor of enabling it for tests too tbh

@adklempner wdyt?

Reply:
Sounds beneficial but warrants a separate PR imo

Reply:
^ agree